manticore/native/cpu/arm.py

```diff
@@ -20,35 +23,108 @@ def HighBit(n):
     return Bit(n, 31)
 
 
-def instruction(body):
-    @wraps(body)
-    def instruction_implementation(cpu, *args, **kwargs):
-        ret = None
+ARMV7_CPU_ADDRESS_BIT_SIZE = 32
+
+
+def instruction(body=None, *, can_take_denormalized_mod_imm: bool = False):
+    """
+    This decorator is used to annotate Armv7Cpu methods as
+    instruction-implementing methods.
+
+    This centralizes some common ARM-specific logic about CPU flags in one place.
+
+    Additionally, this optionally adds /modified immediate normalization/ logic
+    to the wrapped method.  It turns out that Capstone will sometimes disassemble
+    an ARM instruction that take an immediate constant value as its final operand
+    into /two/ immediate operand values, explicitly representing the immediate
+    constant as an 8-bit unsigned number and a 4-bit rotation value.  The
+    modified immediate normalization logic that this decorator adds will convert
+    the explicitly-represented unsigned number and rotation into a single
+    immediate operand-like value that has the appropriate integer value.
+
+    See https://stackoverflow.com/questions/3888158/making-decorators-with-optional-arguments
+    for some decorator-fu specific to making this work both when used both like
+    `@instruction` and like `@instruction(can_take_denormalized_mod_imm=True)`.
+    """
 
-        should_execute = cpu.should_execute_conditional()
+    def decorator(body):
+        if can_take_denormalized_mod_imm:
+            # Need to possibly normalize a modified immediate argument that's
+            # been explicitly split by Capstone into (number, rotation)
+            # components.
+
+            body_sig = inspect_signature(body)
+            num_body_params = len(body_sig.parameters) - 1
+            assert num_body_params > 0
```

Comment:
Why do we subtract `1` on line 57? Is it due to the `cpu` parameter that is present in all instructions?

Reply:
Yeah, you're spot on, it's to account for the `cpu` parameter (which would be called `self` in typical Python code).

Let me see if I can simplify this, or at least add a clarifying comment.

Reply:
Perfect! A clarifying comment would be great :)

Thank you!